packages/expo-build-properties/src/android.ts

```diff
@@ -261,3 +266,32 @@ export const withAndroidQueries: ConfigPlugin<PluginConfigType> = (config, props
     return config;
   });
 };
+
+export const withAndroidDayNightTheme: ConfigPlugin<PluginConfigType> = (config, props) => {
+  return withAndroidStyles(config, (config) => {
+    if (!props.android?.useDayNightTheme) {
+      return config;
+    }
+
+    const { style = [] } = config.modResults.resources;
+    if (!style.length) {
+      return config;
+    }
+
+    // Remove the hardcoded colors.
+    const toRemove = ['android:textColor', 'android:editTextStyle'];
+
+    // We only want to return the AppTheme here. We can drop the `ResetEditText` style.
+    config.modResults.resources.style = [
```

Comment:
please add unit test. we should handle some cases when people adding other styles in their styles.xml. the `config.modResults.resources.style = [` will overwrite everything.